hadoop-plugin/src/main/groovy/com/linkedin/gradle/hadoopValidator/HadoopValidatorPlugin.groovy

```diff
@@ -0,0 +1,82 @@
+/*
+ * Copyright 2015 LinkedIn Corp.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+package com.linkedin.gradle.hadoopValidator;
+
+
+import com.linkedin.gradle.hadoopValidator.PigValidator.PigValidatorPlugin;
+
+import org.gradle.api.Plugin;
+import org.gradle.api.Project;
+import org.gradle.api.UnknownTaskException;
+
+
+/**
+ * HadoopValidatorPlugin is the class that provides a Gradle Plugin which checks for Hadoop ecosystem applications
+ * like Apache Pig.*/
+class HadoopValidatorPlugin implements Plugin<Project> {
+  Project project;
+  Properties properties;
+
+  /**
+   * Applies the Hadoop Validator Plugin, which in turn applies the PigValidator plugin.
+   *
+   * @param project The Gradle project
+   */
+  @Override
+  void apply(Project project) {
+    this.project = project;
+
+    if (project.hasProperty("disableHadoopValidatorPlugin")) {
+      return;
+    }
+    readValidatorProperties();
+
+
+    PigValidatorPlugin plugin = getPigValidatorPlugin().newInstance();
+    plugin.properties = this.properties;
+    plugin.apply(project);
+    try {
+      project.tasks.create(name: "HadoopValidate", group: "Hadoop Plugin",
```

Comment:
Pratik - I would all the tasks to start with a lowercase letter. Can we rename the tasks to be "hadoopValidate" and "pigValidate"?